tests/functions/module_lookup.test.ts

```diff
@@ -1922,3 +1922,344 @@ describe("OFFSET formula", () => {
+    };
+    expect(evaluateCell("D5", grid)).toBe("#ERROR");
+  });
+  test("correct with an empty cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1",
+      A2: "A2", B2: "B2",
+                                    D5: "=ARRAYTOTEXT(A1:B2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("A1,,A2,B2");
+  });
+  test("correct with links, booleans, decimals and errors", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "www.odoo.com", B1: "true", C1: "12/12/2012",
+      A2: "3.14", B2: "#ERROR", C2: "",
+                                    D5: "=ARRAYTOTEXT(A1:C2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("www.odoo.com,TRUE,41255,3.14,#ERROR,");
+  });
+});
+
+describe("FORMULATEXT formula", () => {
+  test("correct with a formula", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "=SUM(B1:B2)", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("=SUM(B1:B2)");
+  });
+  test("correct with a classic cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("A1");
+  });
+  test("correct with an empty cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("");
+  });
+});
+describe("TAKE formula", () => {
+  test("correct for one row", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=TAKE(A1:B2,1)", E5: "", 
+    };
+    //prettier-ignore
+    const result = {
+      A1: "A1", B1: "B1", 
+      A2: "A2", B2: "B2", 
+                                    D5: "A1", E5: "B1",
+    };
+    expect(evaluateGrid(grid)).toEqual(result);
+  });
+  test("correct for -1 column", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=TAKE(A1:B2,2,-1)",
+                                    D6: ""
+    };
+    //prettier-ignore
+    const result = {
+      A1: "A1", B1: "B1", 
+      A2: "A2", B2: "B2", 
+                                    D5: "B1",
+                                    D6: "B2"
+    };
+    expect(evaluateGrid(grid)).toEqual(result);
+  });
+  test("correct for -1 row", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=TAKE(A1:B2,-1)", E5: "", 
+    };
+    //prettier-ignore
+    const result = {
+      A1: "A1", B1: "B1", 
+      A2: "A2", B2: "B2", 
+                                    D5: "A2", E5: "B2",
+    };
+    expect(evaluateGrid(grid)).toEqual(result);
```

Comment:
```suggestion
    const expectedResult = {D5: "A2", E5: "B2",};
    expect(evaluateGrid(grid)).toMatchObject(expectedResult);
```

nitpick, I'd test only the relevant cells, we don't really need to test that A1:B2 is unchanged, it pollutes the test a bit.  Same comment for the other tests